### ISSUE TITLE
 'Newtonsoft' could not be found for \Application.csproj

### ISSUE THREAD

**Issue by NSP-0123456** [microsoft/XmlNotepad#384]
Compilation issue from fresh repo clone.
did a nuget restore before.
I'm trying to compile xml notepad in windows 11 with  Visual Studio 2022 v17.9.6.
I get an error :
 error CS0246: The type or namespace name 'Newtonsoft' could not be found (are you missing a using directive or
an assembly reference?) [C:\dev\DevPerso\XmlNotepad\src\Application\Application.csproj]

Newtonsoft.json is in the package folder but compilation fails.

**Comment by lovettchris** [comment]
It is not entirely a dotnet core solution, so you have to do this:

```dos
nuget  restore xmlnotepad.sln
echo namespace XmlNotepad { public partial class AppAnalytics { private const string ApiKey="%XMLNOTEPAD_ANALYTICSKEY%"; } } > src\model\ApiKey.cs
msbuild xmlnotepad.sln
```

I added this to the build instructions.

**Comment by NSP-0123456** [comment]
Working fine in debug, but still fails in release and just using build.cmd

**Comment by lovettchris** [comment]
Try loading C:\dev\DevPerso\XmlNotepad\src\XmlNotepad.sln and select Release/Any CPU build configuration then select "Build/Rebuild" option.  Rebuild should restore all missing packages.

**Comment by NSP-0123456** [comment]
It is exactly what is not working. even on your build script debug is OK release fails.

**Comment by lovettchris** [comment]
Hmmm, can you run `git clean -dfx` in the root of your XmlNotepad repo then turn on msbuild diagnostic verbosity:

![image](https://github.com/microsoft/XmlNotepad/assets/18707114/e5f886aa-0602-4f98-b362-0f5642be308b)

this select the "Rebuild" option and send me the verbose build output? my electronic mail alias at microsoft is clovett.

**Comment by lovettchris** [comment]
Also if you run this:
```
nuget search Newtonsoft.json
```
Does it show 
```
====================
Source: nuget.org
--------------------
> Newtonsoft.Json | 13.0.3 | Downloads: 4,627,283,248
  Json.NET is a popular high-performance JSON framework for .NET
```
?  You might have a corporate environment that is limiting which packages you can use...

**Comment by lovettchris** [comment]
Closing the issue as I am not able to reproduce this and I haven't heard anything since April.  I can run publish.cmd just fine on multiple machines, even in a brand new [devbox](https://devbox.microsoft.com/).  Note that you do need to install nuget.exe from nuget.org.

**Comment by clovett** [comment]
Oh, hey I just got this problem on another machine and I figured out why it was building ok before.  If I have many packages cached in %USERPROFILE%\.nuget\packages then any mismatching version numbers can be found there and I found that the Newtonsoft assembly is actually version 13.0.0.0 not 13.0.3.0 and I had a similar versioning issue with System.Security.AccessControl.  Once I fixed these the rebuild is clean.
